zenodo/modules/deposit/views.py

```diff
@@ -259,6 +260,18 @@ def current_datetime():
         'current_time': now.time(),
     }
 
+@blueprint.route(
+    '/record/extractmetadata/<key>/<version_id>',
```

Comment:
Since this is already a POST-only endpoint perhaps we can pass the "version_id" in POST data?
i.e. this route would be only: '/record/extractmetadata/'.